dpd-api/src/lib.rs

```diff
@@ -2075,3 +2256,218 @@ pub struct MulticastUnderlayGroupIpParam {
 pub struct MulticastGroupIdParam {
     pub group_id: Option<mcast::MulticastGroupId>,
 }
+
+/// The Physical Coding Sublayer (PCS) counters for a specific link.
+#[derive(Clone, Debug, Deserialize, JsonSchema, Serialize)]
+pub struct LinkPcsCounters {
+    /// The switch port ID.
+    pub port_id: PortId,
+    /// The link ID.
+    pub link_id: LinkId,
+    /// The PCS counter data.
+    pub counters: PcsCounters,
+}
+
+/// The FEC counters for a specific link, including its link ID.
+#[derive(Clone, Debug, Deserialize, JsonSchema, Serialize)]
+pub struct LinkFecRSCounters {
+    /// The switch port ID.
+    pub port_id: PortId,
+    /// The link ID.
+    pub link_id: LinkId,
+    /// The FEC counter data.
+    pub counters: FecRSCounters,
+}
```

Comment:
You may want to put these in dpd-types, especially if they're going to be `replace`d in the future.